crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -248,6 +250,7 @@ impl<'a> Checker<'a> {
             // Set the feature flag to test `TYPE_CHECKING` semantic changes
             semantic.flags |= SemanticModelFlags::NEW_TYPE_CHECKING_BLOCK_DETECTION;
         }
+        let target_version = settings.resolve_target_version(path);
```

Comment:
It could make sense to explicitly pass the `PythonVersion` as an argument and instead compute it once inside `lint_path` (or whatever the function is called that finally calls into `Checker`, there are too many of them)